taf/updater/updater_pipeline.py

```diff
@@ -0,0 +1,1076 @@
+            )
+        return self._output
+
+    @log_on_start(
+        INFO, "Cloning repository and running TUF updater...", logger=taf_logger
+    )
+    @cleanup_decorator
+    def clone_remote_and_run_tuf_updater(self):
+        settings.update_from_filesystem = self.update_from_filesystem
+        settings.conf_directory_root = self.conf_directory_root
+        git_updater = None
+        try:
+            self.state.auth_commits_since_last_validated = None
+            self.state.existing_repo = (
+                Path(self.clients_auth_library_dir, self.state.auth_repo_name).exists()
+                if self.state.auth_repo_name is not None
+                else UpdateStatus.SUCCESS
+            )
+
+            # Clone the validation repository in temp.
+            self.state.auth_repo_name = _clone_validation_repo(
+                self.url, self.state.auth_repo_name
+            )
+            git_updater = GitUpdater(
+                self.url, self.clients_auth_library_dir, self.state.auth_repo_name
+            )
+            self.state.users_auth_repo = git_updater.users_auth_repo
+            _run_tuf_updater(git_updater)
+            self.state.existing_repo = self.state.users_auth_repo.is_git_repository_root
+            self.state.validation_auth_repo = git_updater.validation_auth_repo
+            self._validate_out_of_band_and_update_type()
+
+            self.state.auth_commits_since_last_validated = list(git_updater.commits)
+            self.state.event = (
+                Event.CHANGED
+                if len(self.state.auth_commits_since_last_validated) > 1
+                else Event.UNCHANGED
+            )
+
+            # used for testing purposes
+            if settings.overwrite_last_validated_commit:
+                self.state.last_validated_commit = settings.last_validated_commit
+            else:
+                self.state.last_validated_commit = (
+                    self.state.users_auth_repo.last_validated_commit
+                )
+
+            return UpdateStatus.SUCCESS
+
+        except Exception as e:
+            self.state.error = e
+            self.state.users_auth_repo = None
+
+            if self.state.auth_repo_name is not None:
+                self.state.users_auth_repo = AuthenticationRepository(
+                    self.clients_auth_library_dir,
+                    self.state.auth_repo_name,
+                    urls=[self.url],
+                    conf_directory_root=self.conf_directory_root,
+                )
+            self.state.event = Event.FAILED
+            return UpdateStatus.FAILED
+        finally:
+            # always clean up repository updater
+            if git_updater is not None:
+                git_updater.cleanup()
+
+    @log_on_start(
+        INFO, "Validating out of band commit and update type", logger=taf_logger
+    )
+    def _validate_out_of_band_and_update_type(self):
```

Comment:
I ran `taf repo update --url https://github.com/openlawlibrary/open-law` to test out clone for cloning dependencies as well. It mostly worked, but got the following errors at the end. Here's the interesting output:

```
Validating out of band commit and update type


Update of test/test failed. One or more referenced authentication repositories could not be validated:
 'NoneType' object is not subscriptable
'NoneType' object is not subscriptable
'NoneType' object is not subscriptable
'NoneType' object is not subscriptable
'NoneType' object is not subscriptable
'NoneType' object is not subscriptable
'NoneType' object is not subscriptable


Called LifecycleStage.REPO handler. Event: failed


Called LifecycleStage.UPDATE handler. Event: failed


Update of test/test failed due to error: Update of test/test failed. One or more referenced authentication repositories could not be validated:
 'NoneType' object is not subscriptable
'NoneType' object is not subscriptable
'NoneType' object is not subscriptable
'NoneType' object is not subscriptable
'NoneType' object is not subscriptable
'NoneType' object is not subscriptable
'NoneType' object is not subscriptable
```

Reply:
Fixed